hashicups/resource_order.go

```diff
@@ -259,14 +272,14 @@ func (r resourceOrder) Update(ctx context.Context, req tfsdk.UpdateResourceReque
 		})
 	}
 
-	// Generate resource state struct
-	var result = Order{
+	// Wrap the results from the order in the Terraform schema with the items and new computed values
+	result := Order{
 		ID:          types.String{Value: strconv.Itoa(order.ID)},
 		Items:       ois,
 		LastUpdated: types.String{Value: string(time.Now().Format(time.RFC850))},
 	}
 
-	// Set state
+	// Write the response result variable to state
 	diags = resp.State.Set(ctx, result)
```

Comment:
showing my ignorance here.. but how come for this, we can set it directly to result, but for the others we do `& resourceData`. is there a difference? which one is preferable?

Reply:
Thanks for the explanation!